api/notd/token_listing_processor.py

```diff
@@ -97,3 +100,48 @@ async def get_opensea_listings_for_tokens(self, registryAddress: str, tokenIds:
             # NOTE(krishan711): sleep to avoid opensea limits
             await asyncio.sleep(0.1)
         return listings
+
+    async def get_looks_rare_listings_for_collection(self, registryAddress: str) -> List[RetrievedTokenListing]:
+        listings = []
+        queryData = {
+            'isOrderAsk':'true',
+            'collection':registryAddress,
+            'status[]':'VALID',
+            'pagination[first]': 100,
+            'sort':'PRICE_DESC',
+        }
+        flag = True
+        assetListings = []
+        while flag:
+            response = await self.requester.get(url='https://api.looksrare.org/api/v1/orders', dataDict=queryData, timeout=30)
+            responseJson = response.json()
+            if len(responseJson['data']) == 0:
+                flag = False
+                break
+            for order in responseJson['data']:
+                startDate = datetime.datetime.utcfromtimestamp(order["startTime"])
+                endDate = datetime.datetime.utcfromtimestamp(order["endTime"])
+                currentPrice = int(order["price"])
+                offererAddress = order['signer']
+                sourceId = order["hash"]
+                # NOTE(Femi-Ogunkola): LooksRare seems to send eth listings with weth currency address
+                isValueNative = order["currencyAddress"] == "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2"
+                listing = RetrievedTokenListing(
+                    registryAddress=order['collectionAddress'],
+                    tokenId=order['tokenId'],
+                    startDate=startDate,
+                    endDate=endDate,
+                    isValueNative=isValueNative,
+                    value=currentPrice,
+                    offererAddress=offererAddress,
+                    source='LooksRare',
+                    sourceId=sourceId,
+                )
+                assetListings.append(listing)
+            queryData['pagination[cursor]'] = order['hash']
+        tokenListingDict: Dict[str, RetrievedTokenListing] = defaultdict(RetrievedTokenListing)
+        if len(assetListings) > 0:
+            for listing in assetListings:
+                tokenListingDict[listing.tokenId] = listing
```

Comment:
you haven't implemented the ordering yet?

Reply:
The results are ordered from the api call

Reply:
ah cool